app/components/WalletTextInput.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from 'react'
+import { TextInputProps } from 'react-native'
+import { ThemedView, ThemedText, ThemedTextInput, ThemedTouchableOpacity, ThemedIcon, ThemedSectionTitle } from '@components/themed'
+import { tailwind } from '@tailwind'
+
+type WalletTextInputProps = React.PropsWithChildren<TextInputProps> & IWalletTextInputProps
+type InputType = 'default' | 'numeric'
+
+interface IWalletTextInputProps {
+  inputType: InputType
+  title?: string
+  valid?: boolean
+  inlineValidationText?: string
+  displayClearButton?: boolean
+  onClearButtonPress?: () => void
+  displayFocusStyle?: boolean
+}
+
+export function WalletTextInput (props: WalletTextInputProps): JSX.Element {
+  const [isFocus, setIsFocus] = useState(false)
+  const {
+    inputType,
+    title,
+    valid,
+    inlineValidationText,
+    displayClearButton = false,
+    onClearButtonPress,
+    children,
+    style,
+    ...otherProps
+} = props
+
+  const isInvalid = (): boolean => {
+    return valid === false
```

Comment:
Can we just inline `valid`?

Reply:
Can I keep this for readability purpose? This is due to this function is used in nested ternary operator in theme styling. 

Use case:
``light={tailwind(`bg-white ${isInvalid() ? 'border-error-500' : (isFocus ? 'border-primary-300' : 'border-gray-300')}`)}``

Reply:
But it can be `!valid` right?

Reply:
updated